### ISSUE TITLE
api platform and symfony 4 and mongodb  : The repository for "App\Document\User" must be an instance of "Doctrine\ODM\MongoDB\Repository\DocumentRepository".

### ISSUE THREAD

**Issue by Belssem** [api-platform/core#2617]
i created a document user : 
<?php

namespace App\Document;

use ApiPlatform\Core\Annotation\ApiResource;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;
use DateTime;
use Doctrine\ORM\Mapping as ORM;

/**
 *
 * @ApiResource()
 * @MongoDB\Document
 *
 *
 */
 class User
{
    /** @MongoDB\Id*/
    private $id;

    /** @MongoDB\Field(type="string") */
    private $firstname;

    /** @MongoDB\Field(type="string") */
    private $lastname;

    /** @MongoDB\Field(type="string") */
    private $email;

    /** @MongoDB\Field(type="string") */
    private $login;
    /** @MongoDB\Field(type="string") */
    private $password;
    /** @MongoDB\Field(type="int") */
    private $salary;

    /** @MongoDB\Field(type="date") */
    private $started;

    /** @MongoDB\Field(type="date") */
    private $left;

    /** @MongoDB\ReferenceMany() */
    private $commands;
     /** @MongoDB\EmbedOne(targetDocument="Address") */
     private $address;

     /**
      * @MongoDB\EmbedMany(targetDocument="Role")
      */
     private $Role;

     /**
      * @return mixed
      */
     public function getRole()
     {
         return $this->Role;
     }

     /**
      * @param mixed $Role
      */
     public function setRole($Role): void
     {
         $this->Role = $Role;
     }

    public function  __construct() { $this->commands = new ArrayCollection(); }

    public function getCommands() { return $this->commands; }
    public function addCommands(Command $comm) { $this->commands[] = $comm; }

    /**
     * @return mixed
     */
    public function getEmail()
    {
        return $this->email;
    }

    /**
     * @param mixed $email
     */
    public function setEmail($email): void
    {
        $this->email = $email;
    }

    /**
     * @return mixed
     */
    public function getPassword()
    {
        return $this->password;
    }

    /**
     * @param mixed $password
     */
    public function setPassword($password): void
    {
        $this->password = $password;
    }

    /**
     * @return mixed
     */
    public function getLogin()
    {
        return $this->login;
    }

    /**
     * @param mixed $login
     */
    public function setLogin($login): void
    {
        $this->login = $login;
    }

    /**
     * @return mixed
     */
    public function getLastname()
    {
        return $this->lastname;
    }

    /**
     * @param mixed $lastname
     */
    public function setLastname($lastname): void
    {
        $this->lastname = $lastname;
    }

    public function getId()
    {
        return $this->id;
    }

    public function getChanges()
    {
        return $this->changes;
    }

    public function incrementChanges()
    {
        $this->changes++;
    }

    public function getFirstname()
    {
        return $this->firstname;
    }

    public function setFirstname($firstname)
    {
        $this->firstname = $firstname;
    }

    public function getSalary()
    {
        return $this->salary;
    }

    public function setSalary($salary)
    {
        $this->salary = (int)$salary;
    }

    public function getStarted()
    {
        return $this->started;
    }

    public function setStarted(DateTime $started)
    {
        $this->started = $started;
    }

    public function getLeft()
    {
        return $this->left;
    }

    public function setLeft(DateTime $left)
    {
        $this->left = $left;
    }

    public function getAddress()
    {
        return $this->address;
    }

    public function setAddress(Address $address)
    {
        $this->address = $address;
    }

     /**
      * @param mixed $commands
      */
     public function setCommands($commands): void
     {
         $this->commands = $commands;
     }

     /**
      * @param mixed $id
      */
     public function setId($id): void
     {
         $this->id = $id;
     }

}
and i'm using api platform but when i try to navigate to /api/users i got this error 
The repository for "App\Document\User" must be an instance of "Doctrine\ODM\MongoDB\Repository\DocumentRepository".
 Notice that i didn't have any repository

**Comment by alanpoulain** [comment]
Have you read the documentation? https://api-platform.com/docs/core/mongodb/

**Comment by Belssem** [comment]
yes i read it but juste i didn't use docker i skipped that step and i moved on creating my document 
this is my User document  : 
 <?php

namespace App\Document;

use ApiPlatform\Core\Annotation\ApiResource;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;
use DateTime;
use Doctrine\ORM\Mapping as ORM;


/**
 *
 * @ApiResource()
 * @MongoDB\Document
 *
 *
 */
 class User
{
    /** @MongoDB\Id*/
    private $id;

    /** @MongoDB\Field(type="string") */
    private $firstname;

    /** @MongoDB\Field(type="string") */
    private $lastname;

    /** @MongoDB\Field(type="string") */
    private $email;

    /** @MongoDB\Field(type="string") */
    private $login;
    /** @MongoDB\Field(type="string") */
    private $password;
    /** @MongoDB\Field(type="int") */
    private $salary;

    /** @MongoDB\Field(type="date") */
    private $started;

    /** @MongoDB\Field(type="date") */
    private $left;


    /** @MongoDB\ReferenceMany() */
    private $commands;
     /** @MongoDB\EmbedOne(targetDocument="Address") */
     private $address;

     /**
      * @MongoDB\EmbedMany(targetDocument="Role")
      */
     private $Role;

     /**
      * @return mixed
      */
     public function getRole()
     {
         return $this->Role;
     }

     /**
      * @param mixed $Role
      */
     public function setRole($Role): void
     {
         $this->Role = $Role;
     }


    public function  __construct() { $this->commands = new ArrayCollection(); }

    public function getCommands() { return $this->commands; }
    public function addCommands(Command $comm) { $this->commands[] = $comm; }



    /**
     * @return mixed
     */
    public function getEmail()
    {
        return $this->email;
    }

    /**
     * @param mixed $email
     */
    public function setEmail($email): void
    {
        $this->email = $email;
    }


    /**
     * @return mixed
     */
    public function getPassword()
    {
        return $this->password;
    }

    /**
     * @param mixed $password
     */
    public function setPassword($password): void
    {
        $this->password = $password;
    }

    /**
     * @return mixed
     */
    public function getLogin()
    {
        return $this->login;
    }

    /**
     * @param mixed $login
     */
    public function setLogin($login): void
    {
        $this->login = $login;
    }


    /**
     * @return mixed
     */
    public function getLastname()
    {
        return $this->lastname;
    }

    /**
     * @param mixed $lastname
     */
    public function setLastname($lastname): void
    {
        $this->lastname = $lastname;
    }


    public function getId()
    {
        return $this->id;
    }

    public function getChanges()
    {
        return $this->changes;
    }

    public function incrementChanges()
    {
        $this->changes++;
    }


    public function getFirstname()
    {
        return $this->firstname;
    }

    public function setFirstname($firstname)
    {
        $this->firstname = $firstname;
    }

    public function getSalary()
    {
        return $this->salary;
    }

    public function setSalary($salary)
    {
        $this->salary = (int)$salary;
    }

    public function getStarted()
    {
        return $this->started;
    }

    public function setStarted(DateTime $started)
    {
        $this->started = $started;
    }

    public function getLeft()
    {
        return $this->left;
    }

    public function setLeft(DateTime $left)
    {
        $this->left = $left;
    }

    public function getAddress()
    {
        return $this->address;
    }

    public function setAddress(Address $address)
    {
        $this->address = $address;
    }

     /**
      * @param mixed $commands
      */
     public function setCommands($commands): void
     {
         $this->commands = $commands;
     }

     /**
      * @param mixed $id
      */
     public function setId($id): void
     {
         $this->id = $id;
     }

}

**Comment by Belssem** [comment]
> yes i read it but juste i didn't use docker i skipped that step and i moved on creating my document
> this is my User document :
> 
> Role; } /** * @param mixed $Role */ public function setRole($Role): void { $this->Role = $Role; } public function __construct() { $this->commands = new ArrayCollection(); } public function getCommands() { return $this->commands; } public function addCommands(Command $comm) { $this->commands[] = $comm; } /** * @return mixed */ public function getEmail() { return $this->email; } /** * @param mixed $email */ public function setEmail($email): void { $this->email = $email; } /** * @return mixed */ public function getPassword() { return $this->password; } /** * @param mixed $password */ public function setPassword($password): void { $this->password = $password; } /** * @return mixed */ public function getLogin() { return $this->login; } /** * @param mixed $login */ public function setLogin($login): void { $this->login = $login; } /** * @return mixed */ public function getLastname() { return $this->lastname; } /** * @param mixed $lastname */ public function setLastname($lastname): void { $this->lastname = $lastname; } public function getId() { return $this->id; } public function getChanges() { return $this->changes; } public function incrementChanges() { $this->changes++; } public function getFirstname() { return $this->firstname; } public function setFirstname($firstname) { $this->firstname = $firstname; } public function getSalary() { return $this->salary; } public function setSalary($salary) { $this->salary = (int)$salary; } public function getStarted() { return $this->started; } public function setStarted(DateTime $started) { $this->started = $started; } public function getLeft() { return $this->left; } public function setLeft(DateTime $left) { $this->left = $left; } public function getAddress() { return $this->address; } public function setAddress(Address $address) { $this->address = $address; } /** * @param mixed $commands */ public function setCommands($commands): void { $this->commands = $commands; } /** * @param mixed $id */ public function setId($id): void { $this->id = $id; } }
